clang/lib/CConv/ConstraintResolver.cpp

```diff
@@ -293,32 +292,15 @@ std::set<ConstraintVariable *>
           return getExprConstraintVars(ASE->getBase());
         }
         // add a VarAtom to UOExpr's PVConstraint, for &
-        T = getExprConstraintVars(UOExpr);
-        if (T.empty()) {
-          // If no constraint vars are found, an empty one must be created.
-          // TODO: can we come up with meaningful names in more cases?
-          std::string Name;
-          if (DeclRefExpr *DRE = dyn_cast<DeclRefExpr>(UOExpr)) {
-            Name = DRE->getDecl()->getNameAsString();
-          } else {
-            Name = "";
-          }
-          CAtoms V;
-          ConstraintVariable *newC =
-              new PointerVariableConstraint(V, UOExpr->getType().getAsString(),
-                                            Name, nullptr, false, false, "");
-          T.insert(newC);
-        }
+        std::set<ConstraintVariable *> T = getExprConstraintVars(UOExpr);
```

Comment:
For sanity, lets add an assert to check that `T.empty()` is false.

Reply:
It looks like this uncovered a problem (I thought I checked for this already, but I must've missed it).

```c
struct A a;
struct A *b = &b;
```

Variable `a` generates an empty set. The output is still correct, but the assertion is violated.

Reply:
Since `a` is an lval, it should probably be generating a PVConstraint with no atoms instead of an empty set of constraints. This could be viewed as a bug in how structs are handled.

Reply:
How would you fix it to handle structs? You are saying we should return a `PVConstraint` with no `CAtom`s?

Reply:
Yes, that's the thought.
```c
struct A a;
int b;
```
These could be handled in essentially the same way.